src/configs/passport.ts

```diff
@@ -2,15 +2,59 @@ import passport from 'passport'
 import { Strategy as JwtStrategy } from 'passport-jwt'
 import { dataSource } from './dbConfig'
 import Profile from '../entities/profile.entity'
-import { JWT_SECRET } from './envConfig'
+import {
+  JWT_SECRET,
+  GOOGLE_CLIENT_ID,
+  GOOGLE_REDIRECT_URL,
+  GOOGLE_CLIENT_SECRET
+} from './envConfig'
 import type { Request } from 'express'
 
+// Google Authencation Strategy  ----
+import { Strategy as GoogleStrategy } from 'passport-google-oauth20'
+import { findOrCreateUser } from '../services/auth.service'
+
+passport.use(
+  new GoogleStrategy(
+    {
+      clientID: GOOGLE_CLIENT_ID,
+      clientSecret: GOOGLE_CLIENT_SECRET,
+      callbackURL: GOOGLE_REDIRECT_URL,
+      scope: ['profile', 'email']
+    },
+    async function (accessToken, refreshToken, profile, done) {
+      try {
+        const user = await findOrCreateUser(profile)
+        done(null, user)
+      } catch (err) {
+        done(err as Error)
+      }
+    }
+  )
+)
+
+passport.serializeUser((user: any, done) => {
```

Comment:
Let's avoid using any type

```suggestion
passport.serializeUser((user: User, done) => {
```

Reply:
if i need to add User type first we need to write a User interface right ?